src/mds/MDLog.cc

```diff
@@ -810,6 +810,11 @@ void MDLog::_maybe_expired(LogSegment *ls, int op_prio)
 {
   if (mds->mdcache->is_readonly()) {
     dout(10) << "_maybe_expired, ignoring read-only FS" <<  dendl;
+    submit_mutex.lock();
+    ceph_assert(expiring_segments.count(ls));
+    expiring_segments.erase(ls);
+    expiring_events -= ls->num_events;
+    submit_mutex.unlock();
```

Comment:
Since the fs is read-only it will skip trimming this LogSegment, so just erase it from the expiring list without actually trimming it won't introduce issues ?

Reply:
In MDSRankDispatcher::tick, it will call MDLog::trim , I think it will try to trim this segment again.

Reply:
If the file system is read-only, I don't think the journal grows out-of-bounds. Does this _really_ need special handling? @orange30 @lxbsz

Reply:
I'm not familiar with how read-only happened and recovered. 
One example(just from the code): when PurgeQueue::_go_readonly is called, will the fs may recovere from read-only?
If the fs may recover, I think the trim may hang on one logsegment.
@lxbsz  any comments?

Reply:
Yeah, I think @vshankar is correct. For readonly case the journal shouldn't grow out-of-bounds. And once it recovered from readonly the normal trimming should continue.

There is one case will case the MDS to be readonly: https://tracker.ceph.com/issues/58082 and only after the MDS restart will it recover in this case, I think it should be the same with other cases.

Reply:
What about PurgeQueue::_go_readonly, will the fs may recover from read-only automatically (not mds restart)?

Reply:
I didn't see any where about the PurgeQueue is recovering it from readonly if I didn't miss something.

Reply:
When the file system is read-only, the fs will cannot recover from read-only, so we donot need special handling.